dashboard/app/controllers/api/v1/sections_controller.rb

```diff
@@ -7,6 +8,9 @@ class Api::V1::SectionsController < Api::V1::JsonApiController
 
   rescue_from ActiveRecord::RecordNotFound do |e|
     if e.model == "Section" && %w(join leave).include?(request.filtered_parameters['action'])
+      if current_user&.persisted? && request.filtered_parameters['action'] == 'join'
```

Comment:
For my own learning: Why use `persisted?` here?

Reply:
Not needed here -  I used it in the other captcha PR b/c that one was very intertwined with the user registration process and we wanted to be sure, but that isn't the case here. I'll go ahead and remove it.